### PR TITLE
Fixed backup import with missing category

### PR DESCRIPTION
**Pull request by i906**

### CODE REVIEW

**Comment by mvarnagiris** [comment]
What was the issue here?

File: financius/src/main/java/com/code44/finance/data/backup/BackupDataImporter.java [hunk 1]
```diff
@@ -202,6 +202,8 @@ private void importTransactions(JsonObject json) {
             model.setIncludeInReports(modelJson.get("include_in_reports").getAsBoolean());
             if (model.getTransactionType() == TransactionType.Transfer) {
                 model.setCategory(null);
+            } else {
+                if (model.getCategory() == null) model.setCategory(category);
```
Comment (mvarnagiris) [inline financius/src/main/java/com/code44/finance/data/backup/BackupDataImporter.java]: should follow if () {} else if () {} structure. But I don't see why it's necessary

**Comment by i906** [comment]
During the import process, the category is not set after a transfer transaction. The category remains null after that.

**Comment by mvarnagiris** [comment]
Ahh good spot! Fix that if-else if structure and I will merge

**Comment by i906** [comment]
My 2000 transactions category went missing, so I thought I must find a fix!

File: financius/src/main/java/com/code44/finance/data/backup/BackupDataImporter.java [hunk 2]
```diff
@@ -202,6 +202,10 @@ private void importTransactions(JsonObject json) {
             model.setIncludeInReports(modelJson.get("include_in_reports").getAsBoolean());
             if (model.getTransactionType() == TransactionType.Transfer) {
                 model.setCategory(null);
+            } else {
```
Comment (mvarnagiris) [inline financius/src/main/java/com/code44/finance/data/backup/BackupDataImporter.java]:
```
} else if (model.getCategory() == null) {
    model.setCategory(category);
}
```

**Pull request merged**